policy-controller/src/admission.rs

```diff
@@ -227,13 +236,20 @@ impl Validate<AuthorizationPolicySpec> for Admission {
             bail!("only a single NetworkAuthentication may be set");
         }
 
-        if mtls_authns_count + net_authns_count < spec.required_authentication_refs.len() {
+        if mtls_authns_count + sa_authns_count > 1 {
+            bail!("only MeshTLSAuthentication or ServiceAccount may be set");
```

Comment:
This error message is not clear to me... It seems to insinuate that a NetworkAuthentication can't be set.

```suggestion
            bail!("a MeshTLSAuthentication and ServiceAccount may not be set together");
```